index.qmd

```diff
@@ -0,0 +1,234 @@
+---
+title: Virtual Environments
+date: "`r format(Sys.time(), '%d %B, %Y')`"
+format:
+  html:
+    toc: true
+    theme: www/urbn.scss
+    number-sections: true
+editor: visual
+execute:
+  echo: true
+  eval: false
+---
+
+This guide is intended to help R and Python users at the Urban Institute start using virtual environments. This guide assumes familiarity with Git and GitHub (see [this guide](https://ui-research.github.io/urbngit/) if you are not familiar with these tools) and is focused on helping folks start using virtual environments without having to meaningfully modify their existing workflows. 
+
+## Why should I use a virtual environment? 
+
+Virtual environments promote reproducibility by letting you let you specify project-specific versions of packages. They accomplish this by making it easy to take a snapshot of the version of packages used in a project and restore that snapshot on other computers. They also make it easy to switch between snapshots on a single computer as you switch between projects.  
+
+To make this more concrete, here's a few situations when setting up a virtual environment can save you several hours (or even days!) of frustration: 
+
+* You published an analysis based on 2021 data. A year later, you want to update the analysis with 2022 data, but your code no longer works because a function in `dplyr` was updated. 
+* You need to update your version of `pandas` for a particular project, but a different project requires an older version of the library. 
+* Your coworker is running into errors running your code because of differences in package versions between your computers. 
+* You need to use Python 2.7 for a legacy project, while using Python 3.7 for all other projects. 
+* You want to use a new library that has known compatibility issues with Python 3.8 on Windows computers, so you want to use Python 3.7 for just one specific project. 
+* You got a new computer and don't want to spend hours remembering which R packages to install while resolving `Error in library(<>): there is no package called ‘<>’` errors. 
+* Your code takes a long time to run locally, so you want to leverage powerful [virtual computers in the cloud](http://tech-tools.urban.org/ec2-submission/) – or scale to use tens or hundreds or computers simultaneously – but you first need to install the relevant packages on the virtual machine(s). 
```

Comment:
Very open to adding other examples (or cutting back if this seems too long), but I came up with this list based on personal experience/trauma.

Reply:
These are great!